graylog2-server/src/main/java/org/graylog2/periodical/ContentPackLoaderPeriodical.java

```diff
@@ -89,37 +112,112 @@ protected Logger getLogger() {
+            } catch (MongoException e) {
+                LOG.error("Error while inserting content pack " + file + " into database. Skipping.", e);
+                continue;
+            }
+
+            contentPacks.put(fileName, insertedContentPack);
+        }
+
+        final Set<String> loadedContentPacks = ImmutableSet.<String>builder()
+                .addAll(contentPackLoaderConfig.loadedContentPacks())
+                .addAll(contentPacks.keySet())
+                .build();
+        final Set<String> appliedContentPacks = new HashSet<>(contentPackLoaderConfig.appliedContentPacks());
+
+        LOG.debug("Applying selected content packs");
+        for (Map.Entry<String, ConfigurationBundle> entry : contentPacks.entrySet()) {
+            final String fileName = entry.getKey();
+            final ConfigurationBundle contentPack = entry.getValue();
+
+            if (contentPacksAutoLoad.contains(fileName) && appliedContentPacks.contains(fileName)) {
+                LOG.debug("Content pack {}/{} ({}) already applied. Skipping.", contentPack.getName(), contentPack.getCategory(), fileName);
                 continue;
             }
 
-            if (bundleService.findByNameAndCategory(contentPack.getName(), contentPack.getCategory()) != null) {
-                LOG.debug("Content pack {}/{} already exists in database. Skipping.", contentPack.getCategory(), contentPack.getName());
-            } else {
-                try {
-                    final ConfigurationBundle insertedContentPack = bundleService.insert(contentPack);
-                    LOG.debug("Successfully inserted content pack {} into database with ID {}", resource, insertedContentPack.getId());
-                } catch (MongoException e) {
-                    LOG.error("Error while inserting content pack " + resource + " into database. Skipping.", e);
-                    continue;
+            if (contentPacksAutoLoad.contains(fileName)) {
+                LOG.debug("Applying content pack {}/{} ({})", contentPack.getName(), contentPack.getCategory(), fileName);
+                bundleService.applyConfigurationBundle(contentPack, userService.getAdminUser());
+                appliedContentPacks.add(fileName);
+            }
+        }
+
+        final ContentPackLoaderConfig changedContentPackLoaderConfig =
+                ContentPackLoaderConfig.create(loadedContentPacks, appliedContentPacks);
+        if (!contentPackLoaderConfig.equals(changedContentPackLoaderConfig)) {
+            clusterConfigService.write(changedContentPackLoaderConfig);
+        }
+    }
+
+    private ConfigurationBundle contentPackExists(final ConfigurationBundle contentPack) {
+        final ConfigurationBundle bundle = bundleService.findByNameAndCategory(contentPack.getName(), contentPack.getCategory());
+
+        if (bundle != null) {
+            LOG.debug("Content pack {}/{} already exists in database. Skipping.", contentPack.getCategory(), contentPack.getName());
```

Comment:
I would move the log statement out of this method. This method just does a lookup for the given content pack. It does not decide anything.

Maybe we can just get rid of the method?